src/components/user/repo/user.repo.integration.test.js

```diff
@@ -93,6 +93,39 @@ describe('user repository', () => {
 		});
 	});
 
+	describe('getUsersWithRoles', () => {
+		it('when called with a list of valid userids, then it returns an array of users with populated roles', async () => {
+			const user1 = await testObjects.createTestUser({ roles: ['teacher', 'administrator'] });
+			const user2 = await testObjects.createTestUser({ roles: ['teacher'] });
+			const user3 = await testObjects.createTestUser({ roles: ['teacher'] });
+
+			const result = await userRepo.getUsersWithRoles([user1._id, user2._id, user3._id]);
+
+			expect(result.length).to.equal(3);
+
+			expect(Array.isArray(result[0].roles)).to.be.true;
+			expect(result[0].roles.length).to.equal(2);
+			result[0].roles.forEach((role) => {
+				expect(role).to.haveOwnProperty('name');
```

Comment:
Just a remark, no problem:
There is not check, if the roles returned by the service are correct, just the amount. It feels like it is three tests is one.
But it tests, what is in the description, so no need to change it.